.s2i/bin/run

```diff
@@ -15,18 +15,27 @@ USER_ID=$(id -u)
 GROUP_ID=$(id -g)
 sed "s@${USER_NAME}:x:\${USER_ID}:\${GROUP_ID}@${USER_NAME}:x:${USER_ID}:${GROUP_ID}@g" ${APP_ROOT}/etc/passwd.template > /etc/passwd
 
-INVENTORY="${APP_HOME}/tmp_inventory"
-INVENTORY_ARG="-i ${INVENTORY}"
+INVENTORY="$(mktemp)"
 if [[ -v INVENTORY_FILE ]]; then
-  cp ${INVENTORY_FILE} ${INVENTORY}
+  # If the pointed inventory has execute/search perms we can assume it
+  # contains a dynamic inventory and we use it directly.
+  # Otherwise we make a copy so that ALLOW_ANSIBLE_CONNECTION_LOCAL below
+  # does not attempt to modify the original
+  if [[ -x ${INVENTORY_FILE} ]]; then
```

Comment:
I like this. Could this be extended so have just a single INVENTORY var and have logic to determine if local file (non-executable), local file (executable), or URL (SCHEME://URI)?

That would really simplify the problems mentioned below. Not sure if it covers all use cases.